libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/InputDialog.cs

```diff
@@ -568,7 +568,11 @@ private async Task<InputState> RecognizeInputAsync(DialogContext dc, int turnCou
         private async Task<DialogTurnResult> PromptUserAsync(DialogContext dc, InputState state, CancellationToken cancellationToken = default(CancellationToken))
         {
             var prompt = await this.OnRenderPromptAsync(dc, state, cancellationToken).ConfigureAwait(false);
-            await dc.Context.SendActivityAsync(prompt, cancellationToken).ConfigureAwait(false);
+            if (prompt != null)
```

Comment:
What are the cases where we would not want to throw an exception here?

Reply:
That's a good idea